### PR TITLE
Restructure project

### LINKED ISSUE #17: Rename project
We've decided that the HAPI server and the HAPI adapters for LaTiS are separable and probably belong in different repositories.

This repository is for the HAPI server, so we will rename it `hapi-server` and change the package names so `hapi` is the top-level name. We'll want to also update the description and README to reflect the new focus, namely a HAPI server implemented using LaTiS.

When we begin work on the adapters we'll do that as a subproject in the main LaTiS repository.

### PR DESCRIPTION
**Pull request by lindholc**
At a high level, this commit does the following:

- Renames the project (in the build, repo to follow) to `hapi-server`.
  A name like `latis-hapi` is better suited to a project to develop
  HAPI adapters for LaTiS. This is a project to develop a standalone
  HAPI server and HAPI service interface.

- Updates the README to better reflect that this is a standalone HAPI
  server.

- Renames our base package from `latis.server` to `lasp.hapi` to
  better reflect that this is a standalone HAPI server and not part of
  LaTiS.

- Splits the server stuff into a `lasp.hapi.server` package. This is
  the standalone HAPI server.

- Splits the service interface stuff into a `lasp.hapi.service`
  package. This package could conceivably be made into a separate
  artifact at some point and be used in other servers that require a
  HAPI service interface.

- Splits many things into their own files rather than living where it
  was convenient to put them as they were implemented.

Resolves #17.

### CODE REVIEW

File: src/main/scala/lasp/hapi/server/HapiServer.scala [hunk 1]
```diff
@@ -1,4 +1,5 @@
-package latis.server
+package lasp.hapi
+package server
```
Comment (dlindhol) [inline src/main/scala/lasp/hapi/server/HapiServer.scala]: Why 2 lines for the package?

File: src/main/scala/lasp/hapi/server/HapiServer.scala [hunk 2]
```diff
@@ -11,6 +12,12 @@ import org.http4s._
 import org.http4s.server.middleware.{CORS, CORSConfig}
```
Comment (dlindhol) [inline src/main/scala/lasp/hapi/server/HapiServer.scala]: Any votes for or against this style of import (vs "_" vs separate lines)? I'm not fond of it but I could get used to it.

File: src/main/scala/lasp/hapi/server/HapiServer.scala (hunk 1 above)
Comment (dlindhol) [reply src/main/scala/lasp/hapi/server/HapiServer.scala]: Line too long? ;-)

File: src/main/scala/lasp/hapi/server/HapiServer.scala (hunk 2 above)
Comment (lindholc) [reply src/main/scala/lasp/hapi/server/HapiServer.scala]: I usually prefer multiple lines. However, in this case they are very closely related things, so I guess I went ahead and imported both on the same line.

Perhaps I figured that I wouldn't comment one out without the other. I don't apply this logic consistently.

File: src/main/scala/lasp/hapi/server/HapiServer.scala (hunk 1 above)
Comment (lindholc) [reply src/main/scala/lasp/hapi/server/HapiServer.scala]: This gives me access to both the `lasp.hapi` and `lasp.hapi.server` namespaces, which I found useful for some stuff in `lasp.hapi.service`. I figured I'd just be consistent and do that everywhere. (Except in tests, which need to import things from multiple places less frequently.)

Here is [an article by Odersky about the change](https://www.artima.com/scalazine/articles/chained_package_clauses_in_scala.html). In particular, he says:

> Generally, if your project consists of multiple subpackages, it's a good idea to use a first package clause that names the project as a whole and is followed by a second package clause naming the current subpackage. That way, you can refer to other subpackages of your project without the often long prefix that indicates the project.

I thought I'd give it a try.

**Comment by lindholc** [comment]
I've put everything on a single `package` line to reduce potential confusion. It didn't buy us enough to make it worth that, I suppose.

**Pull request merged**